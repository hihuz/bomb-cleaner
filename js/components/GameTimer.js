import React from 'react';

const GameTimer = props => (
  <div className="game-timer">
    {props.time}
  </div>
);

export default GameTimer;
