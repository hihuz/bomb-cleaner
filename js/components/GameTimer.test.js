import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import GameTimer from './GameTimer';

test('Snapshot', () => {
  const component = shallow(<GameTimer />);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

test('Should display the time passed as prop', () => {
  const expected = '99';
  const component = shallow(<GameTimer time={expected} />);
  const actual = component.find('.game-timer').text();
  expect(actual).toEqual(expected);
});
