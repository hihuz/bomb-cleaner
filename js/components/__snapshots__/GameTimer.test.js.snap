exports[`test Snapshot 1`] = `
<div
  className="game-infos-item">
  <div
    className="game-timer">
    <i
      className="icon-clock-o" />
  </div>
</div>
`;
